dataprofiler/profilers/profile_builder.py

```diff
@@ -1497,23 +1497,81 @@ def save(self, filepath=None):
-    :type samples_per_update: int
-    :param min_true_samples: Minimum number of samples required for the profiler
-    :type min_true_samples: int
-    :param options: Options for the profiler.
-    :type options: ProfilerOptions Object
-    :return: BaseProfiler
-    """
-    # Will want to add 'profiler_type' parameter similar to 'labeler_type'
-    # for specifying structured/unstructured profiler
-
-    # TODO: Add support for creating unstructured profilers via this wrapper
-    return StructuredProfiler(data, samples_per_update, min_true_samples,
-                              options)
+class Profiler(object):
+
+    def __new__(cls, data, samples_per_update=None, min_true_samples=0,
+                options=None, profiler_type=None):
+        """
+        Factory class for instantiating Structured and Unstructured Profilers
+
+        :param data: Data to be profiled
+        :type data: Data class object
+        :param samples_per_update: Number of samples to use to generate profile
+        :type samples_per_update: int
+        :param min_true_samples: Min number of samples required for the profiler
+        :type min_true_samples: int
+        :param options: Options for the profiler.
+        :type options: ProfilerOptions Object
+        :param profiler_type: Type of Profiler ("structured"/"unstructured")
+        :type profiler_type: str
+        :return: BaseProfiler
+        """
+        if profiler_type and profiler_type not in ["structured", "unstructured"]:
+            raise ValueError("Must specify 'profiler_type' to be 'structured' "
+                             "or 'unstructured'.")
+
+        if profiler_type is None:
+            # If type unspecified, and data is Data object, use is_structured
+            if isinstance(data, data_readers.base_data.BaseData):
+                if data.is_structured:
+                    profiler_type = "structured"
+                else:
+                    profiler_type = "unstructured"
+
+            # If type unspecified, and data is Pandas object, need to infer
+            elif isinstance(data, pd.DataFrame) or isinstance(data, pd.Series):
+                # If empty, default to structured
+                if not len(data):
+                    warnings.warn("Empty data given to Profiler, will assume "
+                                  "data is structured, if this is not the "
+                                  "case please specify via 'profiler_type'"
+                                  "kwarg.")
+                    profiler_type = "structured"
+
+                # If multi column df, data is structured
+                elif isinstance(data, pd.DataFrame) and len(data.columns) > 1:
+                    profiler_type = "structured"
+
+                # Single column df or series
+                else:
+                    # Keep a series version of 1 col df for consistency
+                    data_ser = data if isinstance(data, pd.Series) else data[0]
```

Comment:
I'm not sure if by calling data[0] if it creates a copy of the dataset. concern if the dataset was large, but need to look into it.